application/clicommands/JobsCommand.php

```diff
@@ -25,49 +33,124 @@ public function runAction()
         $parallel = (int) $this->Config()->get('scan', 'parallel', 256);
 
         if ($parallel <= 0) {
-            $this->fail("The 'parallel' option must be set to at least 1.");
+            $this->fail("The 'parallel' option must be set to at least 1");
         }
 
         $scheduler = new Scheduler();
+        $this->attachJobsLogging($scheduler);
 
-        $defaultSchedule = $this->Config()->get('jobs', 'default_schedule');
+        $signalHandler = function () use ($scheduler) {
+            $scheduler->removeTasks();
 
-        foreach ($this->Config('jobs') as $name => $jobDescription) {
-            $schedule = $jobDescription->get('schedule', $defaultSchedule);
-
-            if (! $schedule) {
-                Logger::debug("The job '%s' is not scheduled.", $name);
-                continue;
+            Loop::futureTick(function () {
+                Loop::stop();
+            });
+        };
+        Loop::addSignal(SIGINT, $signalHandler);
+        Loop::addSignal(SIGTERM, $signalHandler);
+
+        /** @var Job[] $scheduled Caches scheduled jobs */
+        $scheduled = [];
+        // Periodically check configuration changes to ensure that new jobs are scheduled, jobs are updated,
+        // and deleted jobs are canceled.
+        $watchdog = function () use (&$watchdog, $scheduler, &$scheduled, $parallel) {
+            $jobs = $this->fetchJobs();
+            $outdatedJobs = array_diff_key($scheduled, $jobs);
+            foreach ($outdatedJobs as $job) {
+                Logger::info(
+                    'Removing scheduled job %s, as it either no longer exists in the configuration or its config has' .
```

Comment:
In such cases, don't we put the string concat operator on the next line? Please remove the dot at the end of the string. We had the discussion a few days ago that we don't want to end log messages with a punctuation mark.